tests/unit/pause.tcl

```diff
@@ -359,6 +359,26 @@ start_server {tags {"pause network"}} {
         } {bar2}
     }
 
+    test "Test the randomkey command will not cause the server to get into an infinite loop during the client pause write" {
+        r flushall
+
+        r multi
+        r set key value px 3
+        r client pause 10000 write
+        r exec
+        
+        after 5
+
+        wait_for_condition 50 100 {
+            [r randomkey] == "key"
+        } else {
+            fail "execute randomkey failed, caused by the infinite loop"
+        }
```

Comment:
this `wait_for_condition` isn't necessary, right?
in the past, a single execution of RANDOMKEY would have hang, and now it'll return "key" (but we'd also be ok if it returned `{}`.
i.e. this can be changed to
```suggestion
        set key [r randomkey]
        assert {$key == "key" || $key == ""}
```

Reply:
yes, the first `randomkey` will stuck Redis. and your solution is also reasonable.